src/createServer.js

```diff
@@ -1,3 +1,85 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    // /someText?toCase=SNAKE
+    // /?toCase=SNAKE
+    // console.log('req.url: ', req.url);
+
+    // reading URL
+    const [pathname, queryParams] = req.url.split('?');
+    const text = pathname.slice(1);
+    const params = new URLSearchParams(queryParams);
+    const toCase = params.get('toCase');
+
+    // console.log(typeof toCase, 'TO CASE');
+
+    const errors = [];
+    const caseName = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    if (text.length === 0) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is:'
+        + ' "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (toCase === null) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is:'
+        + ' "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (!caseName.includes(toCase)) {
+      errors.push({
+        message: 'This case is not supported.'
+        + ' Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length > 0) {
+      const errorResponse = {
+        errors,
+      };
+
+      res.setHeader('Content-Type', 'application/json');
+
+      res.statusCode = 400;
+      res.statusText = 'Bad request';
+
+      // console.log('errorResponse: ', errorResponse);
+
+      res.end(JSON.stringify(errorResponse));
+
+      return;
+    }
+
+    const result = convertToCase(text, toCase);
```

Comment:
you can destruct the necessary values here

Reply:
this way looks more  obviously

Reply:
i would use destructuring